File: Py_code/3_M.py
class Solution:
    end = 1
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s) == 0 : return 0
        if len(s) == 1 : return 1
        begin = 0
        max_length = 1
        for self.end in range(1,len(s)):
            if s[self.end] in s[begin:self.end] :
                max_length = max(max_length, self.end - begin)
                while begin != self.end:
                    if s[self.end] not in s[begin:self.end] :
                        break 
                    begin += 1
        max_length = max(max_length, self.end - begin + 1)
        return max_length

File: C++_code/3_M.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        using namespace std;
        int i, end = 1;
        int begin = 0;
        int max_length = 1;
        if (s.length() == 0) return 0;
        if (s.length() == 1) return 1;
        for (end; end < s.length(); end++) {
            for (i = begin; s[i] != s[end]; i++);
            if (i != end) {
                max_length = (end - begin) > max_length ? (end - begin) : max_length;
                begin = i + 1;
            }
        }
        max_length = (end - begin) > max_length ? (end - begin) : max_length;
        return max_length;
    }
};